docs/opensafely-cli.md

```diff
@@ -117,15 +117,75 @@ See the [Codelist](codelist-intro.md) section for more information on codelists.
 
 
 To run your code on your machine, the `opensafely` tool uses the same Docker
-images that run in the secure server environments. These are updated
-periodically, for example when new libraries are installed. If you have error
-messages about missing libraries, your Docker images may need upgrading.
-To pull the most recent Docker images to your machine, run:
+images that run in the secure server environments. There is the
+`cohortextractor` image, for processing study definitions, and then `r`,
+`stata-mp`, and `python` images, for running your analysis code. These last
+three provide a pre-built environment for their specific tool, with a set of libraries.
+
+
+These are updated periodically, for example when new libraries are installed.
+If you have error messages about missing libraries, your Docker images may need
+upgrading.  To pull the most recent Docker images to your machine, run:
 
 ```bash
 opensafely pull
 ```
 
+
+### Developing code with the OpenSAFELY images.
```

Comment:
This heading feels like it implies it's describing the usual way of developing any code. Is that what we're recommending, now? I had thought it was more a debugging tactic, but may have this wrong.

If it's more about debugging alternatives, how about "Using the OpenSAFELY images from the command line" or "Using the OpenSAFELY images outside the pipeline"? And adding an intro sentence along the lines "usually you'll develop your code by running it throuhg the opensafely pipeline [link], but sometimes it's useful to....<blah>"?

Reply:
"Interactive development with the OpenSAFELY environments"?

I think "debugging" is not the right language, as could indicate its very technical rather than generally useful